rclcpp/src/rclcpp/intra_process_manager.cpp

```diff
@@ -220,6 +226,11 @@ IntraProcessManager::can_communicate(
     return false;
   }
 
+  // a publisher and a subscription with different content type can't communicate
+  if (sub_info.is_serialized != pub_info.is_serialized) {
+    return false;
+  }
```

Comment:
Just when I thought I understood what's happening the intraprocess manager, I read this comment and I am confused again. ;-)
I thought the four different variants are so that a non-serialized subscription could obtain values published from a serialized message?

Reply:
@DensoADAS can you comment on this?

Reply:
I think the point here is that [here](https://github.com/DensoADAS/rclcpp/blob/241164204d5fe9fb87a885c25cc3d48c49aef06c/rclcpp/include/rclcpp/publisher.hpp#L135) always the serialized and not serialized publishers are added to the interprocess manager.

Reply:
I think the point here is that here always the serialized and not serialized publishers are added to the interprocess manager.

Reply:
Does this make sense though? Either the publisher is a `serialized` one or not, am I right?
That is, I think we can just only add one id plus the flag whether it's serialized or not.